src/sgame/components/OvermindComponent.cpp

```diff
@@ -72,35 +72,40 @@ Entity* OvermindComponent::FindTarget() {
 	return target;
 }
 
+/**
+ * Find target to attack, which means entity which is:
+ * 1: not on same team
+ * 2: in attack range
+ * 3: the lower health (raw health, armor not considered)
+ * 4: the closer
+ *
+ * a: candidate
+ * b: current target
+ * TODO: make them const (might require deeper changes)
+ **/
 bool OvermindComponent::CompareTargets(Entity& a, Entity& b) const {
+	team_t selfTeam = GetTeamComponent().Team();
 	team_t aTeam = G_Team(a.oldEnt);
 	team_t bTeam = G_Team(b.oldEnt);
 
-	// Prefer humans.
-	if (aTeam == TEAM_HUMANS && bTeam != TEAM_HUMANS) return true;
-	if (aTeam != TEAM_HUMANS && bTeam == TEAM_HUMANS) return false;
+	//don't attack friends or non-players
+	if ( G_OnSameTeam( selfTeam, aTeam ) ) return false;
+	if ( G_OnSameTeam( selfTeam, bTeam ) || !G_IsPlayableTeam( bTeam ) ) return true;
 
+	//check if one is out of range
 	float aDistance = G_Distance(entity.oldEnt, a.oldEnt);
-	float bDistance = G_Distance(entity.oldEnt, b.oldEnt);
-
-	// Rules for humans.
-	if (aTeam == TEAM_HUMANS && bTeam == TEAM_HUMANS) {
-		// Prefer the one in attack range.
-		if (aDistance < ATTACK_RANGE && bDistance > ATTACK_RANGE) return true;
-		if (aDistance > ATTACK_RANGE && bDistance < ATTACK_RANGE) return false;
+	if (aDistance > ATTACK_RANGE) return false;
 
-		HealthComponent* aHealthComponent = a.Get<HealthComponent>();
-		HealthComponent* bHealthComponent = b.Get<HealthComponent>();
+	float bDistance = G_Distance(entity.oldEnt, b.oldEnt);
+	if (bDistance > ATTACK_RANGE) return true;
 
-		// The overmind senses weakness!
-		if (aHealthComponent && bHealthComponent) {
-			if (aHealthComponent->HealthFraction() < bHealthComponent->HealthFraction()) return true;
-			if (aHealthComponent->HealthFraction() > bHealthComponent->HealthFraction()) return false;
-		}
-	}
+	// The overmind senses weakness! (but not armor, so is kind of a joke)
```

Comment:
```suggestion
```

Reply:
thinking anew about it, it also does not use remaining health, so if at some point it have enemies with different amounts of health, the bigger one have 150HP out of 350HP and the 2nd one have 30HP out of 30HP, the bigger one will be targeted, so I indeed think it's not working (generically, that is), even without the armor problem.
This commit is still labelled "bunch of changes" because it needs lot of rework anyway. (I'm going to cleanup that branch to see if some if this stuff can be merged soon)